File: src/extras-kwidgetsaddons/libkmessageboxdontaskagaininterface.cpp
#include <KConfig>
#include <KMessageBoxDontAskAgainInterface>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <kmessageboxdontaskagaininterface.h>
#include "libkmessageboxdontaskagaininterface.h"
#include "libkmessageboxdontaskagaininterface.hxx"

KMessageBoxDontAskAgainInterface* KMessageBoxDontAskAgainInterface_new() {
    return new VirtualKMessageBoxDontAskAgainInterface();
}

void KMessageBoxDontAskAgainInterface_OperatorAssign(KMessageBoxDontAskAgainInterface* self, const KMessageBoxDontAskAgainInterface* param1) {
    self->operator=(*param1);
}

// Derived class handler implementation
bool KMessageBoxDontAskAgainInterface_ShouldBeShownTwoActions(KMessageBoxDontAskAgainInterface* self, const libqt_string dontShowAgainName, int* result) {
    auto* vkmessageboxdontaskagaininterface = dynamic_cast<VirtualKMessageBoxDontAskAgainInterface*>(self);
    QString dontShowAgainName_QString = QString::fromUtf8(dontShowAgainName.data, dontShowAgainName.len);
    if (vkmessageboxdontaskagaininterface && vkmessageboxdontaskagaininterface->isVirtualKMessageBoxDontAskAgainInterface) {
        return vkmessageboxdontaskagaininterface->shouldBeShownTwoActions(dontShowAgainName_QString, (KMessageBox::ButtonCode&)(*result));
    } else {
        return ((VirtualKMessageBoxDontAskAgainInterface*)self)->shouldBeShownTwoActions(dontShowAgainName_QString, (KMessageBox::ButtonCode&)(*result));
    }
}

// Base class handler implementation
bool KMessageBoxDontAskAgainInterface_QBaseShouldBeShownTwoActions(KMessageBoxDontAskAgainInterface* self, const libqt_string dontShowAgainName, int* result) {
    auto* vkmessageboxdontaskagaininterface = dynamic_cast<VirtualKMessageBoxDontAskAgainInterface*>(self);
    QString dontShowAgainName_QString = QString::fromUtf8(dontShowAgainName.data, dontShowAgainName.len);
    if (vkmessageboxdontaskagaininterface && vkmessageboxdontaskagaininterface->isVirtualKMessageBoxDontAskAgainInterface) {
        vkmessageboxdontaskagaininterface->setKMessageBoxDontAskAgainInterface_ShouldBeShownTwoActions_IsBase(true);
        return vkmessageboxdontaskagaininterface->shouldBeShownTwoActions(dontShowAgainName_QString, (KMessageBox::ButtonCode&)(*result));
    } else {
        return ((VirtualKMessageBoxDontAskAgainInterface*)self)->shouldBeShownTwoActions(dontShowAgainName_QString, (KMessageBox::ButtonCode&)(*result));
    }
}

// Auxiliary method to allow providing re-implementation
void KMessageBoxDontAskAgainInterface_OnShouldBeShownTwoActions(KMessageBoxDontAskAgainInterface* self, intptr_t slot) {
    auto* vkmessageboxdontaskagaininterface = dynamic_cast<VirtualKMessageBoxDontAskAgainInterface*>(self);
    if (vkmessageboxdontaskagaininterface && vkmessageboxdontaskagaininterface->isVirtualKMessageBoxDontAskAgainInterface) {
        vkmessageboxdontaskagaininterface->setKMessageBoxDontAskAgainInterface_ShouldBeShownTwoActions_Callback(reinterpret_cast<VirtualKMessageBoxDontAskAgainInterface::KMessageBoxDontAskAgainInterface_ShouldBeShownTwoActions_Callback>(slot));
    }
}

// Derived class handler implementation
bool KMessageBoxDontAskAgainInterface_ShouldBeShownContinue(KMessageBoxDontAskAgainInterface* self, const libqt_string dontShowAgainName) {
    auto* vkmessageboxdontaskagaininterface = dynamic_cast<VirtualKMessageBoxDontAskAgainInterface*>(self);
    QString dontShowAgainName_QString = QString::fromUtf8(dontShowAgainName.data, dontShowAgainName.len);
    if (vkmessageboxdontaskagaininterface && vkmessageboxdontaskagaininterface->isVirtualKMessageBoxDontAskAgainInterface) {
        return vkmessageboxdontaskagaininterface->shouldBeShownContinue(dontShowAgainName_QString);
    } else {
        return ((VirtualKMessageBoxDontAskAgainInterface*)self)->shouldBeShownContinue(dontShowAgainName_QString);
    }
}

// Base class handler implementation
bool KMessageBoxDontAskAgainInterface_QBaseShouldBeShownContinue(KMessageBoxDontAskAgainInterface* self, const libqt_string dontShowAgainName) {
    auto* vkmessageboxdontaskagaininterface = dynamic_cast<VirtualKMessageBoxDontAskAgainInterface*>(self);
    QString dontShowAgainName_QString = QString::fromUtf8(dontShowAgainName.data, dontShowAgainName.len);
    if (vkmessageboxdontaskagaininterface && vkmessageboxdontaskagaininterface->isVirtualKMessageBoxDontAskAgainInterface) {
        vkmessageboxdontaskagaininterface->setKMessageBoxDontAskAgainInterface_ShouldBeShownContinue_IsBase(true);
        return vkmessageboxdontaskagaininterface->shouldBeShownContinue(dontShowAgainName_QString);
    } else {
        return ((VirtualKMessageBoxDontAskAgainInterface*)self)->shouldBeShownContinue(dontShowAgainName_QString);
    }
}

// Auxiliary method to allow providing re-implementation
void KMessageBoxDontAskAgainInterface_OnShouldBeShownContinue(KMessageBoxDontAskAgainInterface* self, intptr_t slot) {
    auto* vkmessageboxdontaskagaininterface = dynamic_cast<VirtualKMessageBoxDontAskAgainInterface*>(self);
    if (vkmessageboxdontaskagaininterface && vkmessageboxdontaskagaininterface->isVirtualKMessageBoxDontAskAgainInterface) {
        vkmessageboxdontaskagaininterface->setKMessageBoxDontAskAgainInterface_ShouldBeShownContinue_Callback(reinterpret_cast<VirtualKMessageBoxDontAskAgainInterface::KMessageBoxDontAskAgainInterface_ShouldBeShownContinue_Callback>(slot));
    }
}

// Derived class handler implementation
void KMessageBoxDontAskAgainInterface_SaveDontShowAgainTwoActions(KMessageBoxDontAskAgainInterface* self, const libqt_string dontShowAgainName, int result) {
    auto* vkmessageboxdontaskagaininterface = dynamic_cast<VirtualKMessageBoxDontAskAgainInterface*>(self);
    QString dontShowAgainName_QString = QString::fromUtf8(dontShowAgainName.data, dontShowAgainName.len);
    if (vkmessageboxdontaskagaininterface && vkmessageboxdontaskagaininterface->isVirtualKMessageBoxDontAskAgainInterface) {
        vkmessageboxdontaskagaininterface->saveDontShowAgainTwoActions(dontShowAgainName_QString, static_cast<KMessageBox::ButtonCode>(result));
    } else {
        ((VirtualKMessageBoxDontAskAgainInterface*)self)->saveDontShowAgainTwoActions(dontShowAgainName_QString, static_cast<KMessageBox::ButtonCode>(result));
    }
}

// Base class handler implementation
void KMessageBoxDontAskAgainInterface_QBaseSaveDontShowAgainTwoActions(KMessageBoxDontAskAgainInterface* self, const libqt_string dontShowAgainName, int result) {
    auto* vkmessageboxdontaskagaininterface = dynamic_cast<VirtualKMessageBoxDontAskAgainInterface*>(self);
    QString dontShowAgainName_QString = QString::fromUtf8(dontShowAgainName.data, dontShowAgainName.len);
    if (vkmessageboxdontaskagaininterface && vkmessageboxdontaskagaininterface->isVirtualKMessageBoxDontAskAgainInterface) {
        vkmessageboxdontaskagaininterface->setKMessageBoxDontAskAgainInterface_SaveDontShowAgainTwoActions_IsBase(true);
        vkmessageboxdontaskagaininterface->saveDontShowAgainTwoActions(dontShowAgainName_QString, static_cast<KMessageBox::ButtonCode>(result));
    } else {
        ((VirtualKMessageBoxDontAskAgainInterface*)self)->saveDontShowAgainTwoActions(dontShowAgainName_QString, static_cast<KMessageBox::ButtonCode>(result));
    }
}

// Auxiliary method to allow providing re-implementation
void KMessageBoxDontAskAgainInterface_OnSaveDontShowAgainTwoActions(KMessageBoxDontAskAgainInterface* self, intptr_t slot) {
    auto* vkmessageboxdontaskagaininterface = dynamic_cast<VirtualKMessageBoxDontAskAgainInterface*>(self);
    if (vkmessageboxdontaskagaininterface && vkmessageboxdontaskagaininterface->isVirtualKMessageBoxDontAskAgainInterface) {
        vkmessageboxdontaskagaininterface->setKMessageBoxDontAskAgainInterface_SaveDontShowAgainTwoActions_Callback(reinterpret_cast<VirtualKMessageBoxDontAskAgainInterface::KMessageBoxDontAskAgainInterface_SaveDontShowAgainTwoActions_Callback>(slot));
    }
}

// Derived class handler implementation
void KMessageBoxDontAskAgainInterface_SaveDontShowAgainContinue(KMessageBoxDontAskAgainInterface* self, const libqt_string dontShowAgainName) {
    auto* vkmessageboxdontaskagaininterface = dynamic_cast<VirtualKMessageBoxDontAskAgainInterface*>(self);
    QString dontShowAgainName_QString = QString::fromUtf8(dontShowAgainName.data, dontShowAgainName.len);
    if (vkmessageboxdontaskagaininterface && vkmessageboxdontaskagaininterface->isVirtualKMessageBoxDontAskAgainInterface) {
        vkmessageboxdontaskagaininterface->saveDontShowAgainContinue(dontShowAgainName_QString);
    } else {
        ((VirtualKMessageBoxDontAskAgainInterface*)self)->saveDontShowAgainContinue(dontShowAgainName_QString);
    }
}

// Base class handler implementation
void KMessageBoxDontAskAgainInterface_QBaseSaveDontShowAgainContinue(KMessageBoxDontAskAgainInterface* self, const libqt_string dontShowAgainName) {
    auto* vkmessageboxdontaskagaininterface = dynamic_cast<VirtualKMessageBoxDontAskAgainInterface*>(self);
    QString dontShowAgainName_QString = QString::fromUtf8(dontShowAgainName.data, dontShowAgainName.len);
    if (vkmessageboxdontaskagaininterface && vkmessageboxdontaskagaininterface->isVirtualKMessageBoxDontAskAgainInterface) {
        vkmessageboxdontaskagaininterface->setKMessageBoxDontAskAgainInterface_SaveDontShowAgainContinue_IsBase(true);
        vkmessageboxdontaskagaininterface->saveDontShowAgainContinue(dontShowAgainName_QString);
    } else {
        ((VirtualKMessageBoxDontAskAgainInterface*)self)->saveDontShowAgainContinue(dontShowAgainName_QString);
    }
}

// Auxiliary method to allow providing re-implementation
void KMessageBoxDontAskAgainInterface_OnSaveDontShowAgainContinue(KMessageBoxDontAskAgainInterface* self, intptr_t slot) {
    auto* vkmessageboxdontaskagaininterface = dynamic_cast<VirtualKMessageBoxDontAskAgainInterface*>(self);
    if (vkmessageboxdontaskagaininterface && vkmessageboxdontaskagaininterface->isVirtualKMessageBoxDontAskAgainInterface) {
        vkmessageboxdontaskagaininterface->setKMessageBoxDontAskAgainInterface_SaveDontShowAgainContinue_Callback(reinterpret_cast<VirtualKMessageBoxDontAskAgainInterface::KMessageBoxDontAskAgainInterface_SaveDontShowAgainContinue_Callback>(slot));
    }
}

// Derived class handler implementation
void KMessageBoxDontAskAgainInterface_EnableAllMessages(KMessageBoxDontAskAgainInterface* self) {
    auto* vkmessageboxdontaskagaininterface = dynamic_cast<VirtualKMessageBoxDontAskAgainInterface*>(self);
    if (vkmessageboxdontaskagaininterface && vkmessageboxdontaskagaininterface->isVirtualKMessageBoxDontAskAgainInterface) {
        vkmessageboxdontaskagaininterface->enableAllMessages();
    } else {
        ((VirtualKMessageBoxDontAskAgainInterface*)self)->enableAllMessages();
    }
}

// Base class handler implementation
void KMessageBoxDontAskAgainInterface_QBaseEnableAllMessages(KMessageBoxDontAskAgainInterface* self) {
    auto* vkmessageboxdontaskagaininterface = dynamic_cast<VirtualKMessageBoxDontAskAgainInterface*>(self);
    if (vkmessageboxdontaskagaininterface && vkmessageboxdontaskagaininterface->isVirtualKMessageBoxDontAskAgainInterface) {
        vkmessageboxdontaskagaininterface->setKMessageBoxDontAskAgainInterface_EnableAllMessages_IsBase(true);
        vkmessageboxdontaskagaininterface->enableAllMessages();
    } else {
        ((VirtualKMessageBoxDontAskAgainInterface*)self)->enableAllMessages();
    }
}

// Auxiliary method to allow providing re-implementation
void KMessageBoxDontAskAgainInterface_OnEnableAllMessages(KMessageBoxDontAskAgainInterface* self, intptr_t slot) {
    auto* vkmessageboxdontaskagaininterface = dynamic_cast<VirtualKMessageBoxDontAskAgainInterface*>(self);
    if (vkmessageboxdontaskagaininterface && vkmessageboxdontaskagaininterface->isVirtualKMessageBoxDontAskAgainInterface) {
        vkmessageboxdontaskagaininterface->setKMessageBoxDontAskAgainInterface_EnableAllMessages_Callback(reinterpret_cast<VirtualKMessageBoxDontAskAgainInterface::KMessageBoxDontAskAgainInterface_EnableAllMessages_Callback>(slot));
    }
}

// Derived class handler implementation
void KMessageBoxDontAskAgainInterface_EnableMessage(KMessageBoxDontAskAgainInterface* self, const libqt_string dontShowAgainName) {
    auto* vkmessageboxdontaskagaininterface = dynamic_cast<VirtualKMessageBoxDontAskAgainInterface*>(self);
    QString dontShowAgainName_QString = QString::fromUtf8(dontShowAgainName.data, dontShowAgainName.len);
    if (vkmessageboxdontaskagaininterface && vkmessageboxdontaskagaininterface->isVirtualKMessageBoxDontAskAgainInterface) {
        vkmessageboxdontaskagaininterface->enableMessage(dontShowAgainName_QString);
    } else {
        ((VirtualKMessageBoxDontAskAgainInterface*)self)->enableMessage(dontShowAgainName_QString);
    }
}

// Base class handler implementation
void KMessageBoxDontAskAgainInterface_QBaseEnableMessage(KMessageBoxDontAskAgainInterface* self, const libqt_string dontShowAgainName) {
    auto* vkmessageboxdontaskagaininterface = dynamic_cast<VirtualKMessageBoxDontAskAgainInterface*>(self);
    QString dontShowAgainName_QString = QString::fromUtf8(dontShowAgainName.data, dontShowAgainName.len);
    if (vkmessageboxdontaskagaininterface && vkmessageboxdontaskagaininterface->isVirtualKMessageBoxDontAskAgainInterface) {
        vkmessageboxdontaskagaininterface->setKMessageBoxDontAskAgainInterface_EnableMessage_IsBase(true);
        vkmessageboxdontaskagaininterface->enableMessage(dontShowAgainName_QString);
    } else {
        ((VirtualKMessageBoxDontAskAgainInterface*)self)->enableMessage(dontShowAgainName_QString);
    }
}

// Auxiliary method to allow providing re-implementation
void KMessageBoxDontAskAgainInterface_OnEnableMessage(KMessageBoxDontAskAgainInterface* self, intptr_t slot) {
    auto* vkmessageboxdontaskagaininterface = dynamic_cast<VirtualKMessageBoxDontAskAgainInterface*>(self);
    if (vkmessageboxdontaskagaininterface && vkmessageboxdontaskagaininterface->isVirtualKMessageBoxDontAskAgainInterface) {
        vkmessageboxdontaskagaininterface->setKMessageBoxDontAskAgainInterface_EnableMessage_Callback(reinterpret_cast<VirtualKMessageBoxDontAskAgainInterface::KMessageBoxDontAskAgainInterface_EnableMessage_Callback>(slot));
    }
}

// Derived class handler implementation
void KMessageBoxDontAskAgainInterface_SetConfig(KMessageBoxDontAskAgainInterface* self, KConfig* config) {
    auto* vkmessageboxdontaskagaininterface = dynamic_cast<VirtualKMessageBoxDontAskAgainInterface*>(self);
    if (vkmessageboxdontaskagaininterface && vkmessageboxdontaskagaininterface->isVirtualKMessageBoxDontAskAgainInterface) {
        vkmessageboxdontaskagaininterface->setConfig(config);
    } else {
        ((VirtualKMessageBoxDontAskAgainInterface*)self)->setConfig(config);
    }
}

// Base class handler implementation
void KMessageBoxDontAskAgainInterface_QBaseSetConfig(KMessageBoxDontAskAgainInterface* self, KConfig* config) {
    auto* vkmessageboxdontaskagaininterface = dynamic_cast<VirtualKMessageBoxDontAskAgainInterface*>(self);
    if (vkmessageboxdontaskagaininterface && vkmessageboxdontaskagaininterface->isVirtualKMessageBoxDontAskAgainInterface) {
        vkmessageboxdontaskagaininterface->setKMessageBoxDontAskAgainInterface_SetConfig_IsBase(true);
        vkmessageboxdontaskagaininterface->setConfig(config);
    } else {
        ((VirtualKMessageBoxDontAskAgainInterface*)self)->setConfig(config);
    }
}

// Auxiliary method to allow providing re-implementation
void KMessageBoxDontAskAgainInterface_OnSetConfig(KMessageBoxDontAskAgainInterface* self, intptr_t slot) {
    auto* vkmessageboxdontaskagaininterface = dynamic_cast<VirtualKMessageBoxDontAskAgainInterface*>(self);
    if (vkmessageboxdontaskagaininterface && vkmessageboxdontaskagaininterface->isVirtualKMessageBoxDontAskAgainInterface) {
        vkmessageboxdontaskagaininterface->setKMessageBoxDontAskAgainInterface_SetConfig_Callback(reinterpret_cast<VirtualKMessageBoxDontAskAgainInterface::KMessageBoxDontAskAgainInterface_SetConfig_Callback>(slot));
    }
}

void KMessageBoxDontAskAgainInterface_Delete(KMessageBoxDontAskAgainInterface* self) {
    delete self;
}

File: src/extras-kwidgetsaddons/libkmessageboxdontaskagaininterface.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");
const kmessagebox_enums = @import("libkmessagebox.zig").enums;

/// https://api.kde.org/kmessageboxdontaskagaininterface.html
pub const kmessageboxdontaskagaininterface = struct {
    /// New constructs a new KMessageBoxDontAskAgainInterface object.
    ///
    ///
    pub fn New() QtC.KMessageBoxDontAskAgainInterface {
        return qtc.KMessageBoxDontAskAgainInterface_new();
    }

    /// [Qt documentation](https://api.kde.org/kmessageboxdontaskagaininterface.html#shouldBeShownTwoActions)
    ///
    /// ``` self: QtC.KMessageBoxDontAskAgainInterface, dontShowAgainName: []const u8, result: *kmessagebox_enums.ButtonCode ```
    pub fn ShouldBeShownTwoActions(self: ?*anyopaque, dontShowAgainName: []const u8, result: *i32) bool {
        const dontShowAgainName_str = qtc.libqt_string{
            .len = dontShowAgainName.len,
            .data = dontShowAgainName.ptr,
        };
        return qtc.KMessageBoxDontAskAgainInterface_ShouldBeShownTwoActions(@ptrCast(self), dontShowAgainName_str, @ptrCast(result));
    }

    /// [Qt documentation](https://api.kde.org/kmessageboxdontaskagaininterface.html#shouldBeShownTwoActions)
    ///
    /// Allows for overriding the related default method
    ///
    /// ``` self: QtC.KMessageBoxDontAskAgainInterface, slot: fn (self: QtC.KMessageBoxDontAskAgainInterface, dontShowAgainName: [*:0]const u8, result: *kmessagebox_enums.ButtonCode) callconv(.c) bool ```
    pub fn OnShouldBeShownTwoActions(self: ?*anyopaque, slot: fn (?*anyopaque, [*:0]const u8, *i32) callconv(.c) bool) void {
        qtc.KMessageBoxDontAskAgainInterface_OnShouldBeShownTwoActions(@ptrCast(self), @as(isize, @bitCast(@intFromPtr(&slot))));
    }

    /// [Qt documentation](https://api.kde.org/kmessageboxdontaskagaininterface.html#shouldBeShownTwoActions)
    ///
    /// Base class method implementation
    ///
    /// ``` self: QtC.KMessageBoxDontAskAgainInterface, dontShowAgainName: []const u8, result: *kmessagebox_enums.ButtonCode ```
    pub fn QBaseShouldBeShownTwoActions(self: ?*anyopaque, dontShowAgainName: []const u8, result: *i32) bool {
        const dontShowAgainName_str = qtc.libqt_string{
            .len = dontShowAgainName.len,
            .data = dontShowAgainName.ptr,
        };
        return qtc.KMessageBoxDontAskAgainInterface_QBaseShouldBeShownTwoActions(@ptrCast(self), dontShowAgainName_str, @ptrCast(result));
    }

    /// [Qt documentation](https://api.kde.org/kmessageboxdontaskagaininterface.html#shouldBeShownContinue)
    ///
    /// ``` self: QtC.KMessageBoxDontAskAgainInterface, dontShowAgainName: []const u8 ```
    pub fn ShouldBeShownContinue(self: ?*anyopaque, dontShowAgainName: []const u8) bool {
        const dontShowAgainName_str = qtc.libqt_string{
            .len = dontShowAgainName.len,
            .data = dontShowAgainName.ptr,
        };
        return qtc.KMessageBoxDontAskAgainInterface_ShouldBeShownContinue(@ptrCast(self), dontShowAgainName_str);
    }

    /// [Qt documentation](https://api.kde.org/kmessageboxdontaskagaininterface.html#shouldBeShownContinue)
    ///
    /// Allows for overriding the related default method
    ///
    /// ``` self: QtC.KMessageBoxDontAskAgainInterface, slot: fn (self: QtC.KMessageBoxDontAskAgainInterface, dontShowAgainName: [*:0]const u8) callconv(.c) bool ```
    pub fn OnShouldBeShownContinue(self: ?*anyopaque, slot: fn (?*anyopaque, [*:0]const u8) callconv(.c) bool) void {
        qtc.KMessageBoxDontAskAgainInterface_OnShouldBeShownContinue(@ptrCast(self), @as(isize, @bitCast(@intFromPtr(&slot))));
    }

    /// [Qt documentation](https://api.kde.org/kmessageboxdontaskagaininterface.html#shouldBeShownContinue)
    ///
    /// Base class method implementation
    ///
    /// ``` self: QtC.KMessageBoxDontAskAgainInterface, dontShowAgainName: []const u8 ```
    pub fn QBaseShouldBeShownContinue(self: ?*anyopaque, dontShowAgainName: []const u8) bool {
        const dontShowAgainName_str = qtc.libqt_string{
            .len = dontShowAgainName.len,
            .data = dontShowAgainName.ptr,
        };
        return qtc.KMessageBoxDontAskAgainInterface_QBaseShouldBeShownContinue(@ptrCast(self), dontShowAgainName_str);
    }

    /// [Qt documentation](https://api.kde.org/kmessageboxdontaskagaininterface.html#saveDontShowAgainTwoActions)
    ///
    /// ``` self: QtC.KMessageBoxDontAskAgainInterface, dontShowAgainName: []const u8, result: kmessagebox_enums.ButtonCode ```
    pub fn SaveDontShowAgainTwoActions(self: ?*anyopaque, dontShowAgainName: []const u8, result: i32) void {
        const dontShowAgainName_str = qtc.libqt_string{
            .len = dontShowAgainName.len,
            .data = dontShowAgainName.ptr,
        };
        qtc.KMessageBoxDontAskAgainInterface_SaveDontShowAgainTwoActions(@ptrCast(self), dontShowAgainName_str, @intCast(result));
    }

    /// [Qt documentation](https://api.kde.org/kmessageboxdontaskagaininterface.html#saveDontShowAgainTwoActions)
    ///
    /// Allows for overriding the related default method
    ///
    /// ``` self: QtC.KMessageBoxDontAskAgainInterface, slot: fn (self: QtC.KMessageBoxDontAskAgainInterface, dontShowAgainName: [*:0]const u8, result: kmessagebox_enums.ButtonCode) callconv(.c) void ```
    pub fn OnSaveDontShowAgainTwoActions(self: ?*anyopaque, slot: fn (?*anyopaque, [*:0]const u8, i32) callconv(.c) void) void {
        qtc.KMessageBoxDontAskAgainInterface_OnSaveDontShowAgainTwoActions(@ptrCast(self), @as(isize, @bitCast(@intFromPtr(&slot))));
    }

    /// [Qt documentation](https://api.kde.org/kmessageboxdontaskagaininterface.html#saveDontShowAgainTwoActions)
    ///
    /// Base class method implementation
    ///
    /// ``` self: QtC.KMessageBoxDontAskAgainInterface, dontShowAgainName: []const u8, result: kmessagebox_enums.ButtonCode ```
    pub fn QBaseSaveDontShowAgainTwoActions(self: ?*anyopaque, dontShowAgainName: []const u8, result: i32) void {
        const dontShowAgainName_str = qtc.libqt_string{
            .len = dontShowAgainName.len,
            .data = dontShowAgainName.ptr,
        };
        qtc.KMessageBoxDontAskAgainInterface_QBaseSaveDontShowAgainTwoActions(@ptrCast(self), dontShowAgainName_str, @intCast(result));
    }

    /// [Qt documentation](https://api.kde.org/kmessageboxdontaskagaininterface.html#saveDontShowAgainContinue)
    ///
    /// ``` self: QtC.KMessageBoxDontAskAgainInterface, dontShowAgainName: []const u8 ```
    pub fn SaveDontShowAgainContinue(self: ?*anyopaque, dontShowAgainName: []const u8) void {
        const dontShowAgainName_str = qtc.libqt_string{
            .len = dontShowAgainName.len,
            .data = dontShowAgainName.ptr,
        };
        qtc.KMessageBoxDontAskAgainInterface_SaveDontShowAgainContinue(@ptrCast(self), dontShowAgainName_str);
    }

    /// [Qt documentation](https://api.kde.org/kmessageboxdontaskagaininterface.html#saveDontShowAgainContinue)
    ///
    /// Allows for overriding the related default method
    ///
    /// ``` self: QtC.KMessageBoxDontAskAgainInterface, slot: fn (self: QtC.KMessageBoxDontAskAgainInterface, dontShowAgainName: [*:0]const u8) callconv(.c) void ```
    pub fn OnSaveDontShowAgainContinue(self: ?*anyopaque, slot: fn (?*anyopaque, [*:0]const u8) callconv(.c) void) void {
        qtc.KMessageBoxDontAskAgainInterface_OnSaveDontShowAgainContinue(@ptrCast(self), @as(isize, @bitCast(@intFromPtr(&slot))));
    }

    /// [Qt documentation](https://api.kde.org/kmessageboxdontaskagaininterface.html#saveDontShowAgainContinue)
    ///
    /// Base class method implementation
    ///
    /// ``` self: QtC.KMessageBoxDontAskAgainInterface, dontShowAgainName: []const u8 ```
    pub fn QBaseSaveDontShowAgainContinue(self: ?*anyopaque, dontShowAgainName: []const u8) void {
        const dontShowAgainName_str = qtc.libqt_string{
            .len = dontShowAgainName.len,
            .data = dontShowAgainName.ptr,
        };
        qtc.KMessageBoxDontAskAgainInterface_QBaseSaveDontShowAgainContinue(@ptrCast(self), dontShowAgainName_str);
    }

    /// [Qt documentation](https://api.kde.org/kmessageboxdontaskagaininterface.html#enableAllMessages)
    ///
    /// ``` self: QtC.KMessageBoxDontAskAgainInterface ```
    pub fn EnableAllMessages(self: ?*anyopaque) void {
        qtc.KMessageBoxDontAskAgainInterface_EnableAllMessages(@ptrCast(self));
    }

    /// [Qt documentation](https://api.kde.org/kmessageboxdontaskagaininterface.html#enableAllMessages)
    ///
    /// Allows for overriding the related default method
    ///
    /// ``` self: QtC.KMessageBoxDontAskAgainInterface, slot: fn () callconv(.c) void ```
    pub fn OnEnableAllMessages(self: ?*anyopaque, slot: fn () callconv(.c) void) void {
        qtc.KMessageBoxDontAskAgainInterface_OnEnableAllMessages(@ptrCast(self), @as(isize, @bitCast(@intFromPtr(&slot))));
    }

    /// [Qt documentation](https://api.kde.org/kmessageboxdontaskagaininterface.html#enableAllMessages)
    ///
    /// Base class method implementation
    ///
    /// ``` self: QtC.KMessageBoxDontAskAgainInterface ```
    pub fn QBaseEnableAllMessages(self: ?*anyopaque) void {
        qtc.KMessageBoxDontAskAgainInterface_QBaseEnableAllMessages(@ptrCast(self));
    }

    /// [Qt documentation](https://api.kde.org/kmessageboxdontaskagaininterface.html#enableMessage)
    ///
    /// ``` self: QtC.KMessageBoxDontAskAgainInterface, dontShowAgainName: []const u8 ```
    pub fn EnableMessage(self: ?*anyopaque, dontShowAgainName: []const u8) void {
        const dontShowAgainName_str = qtc.libqt_string{
            .len = dontShowAgainName.len,
            .data = dontShowAgainName.ptr,
        };
        qtc.KMessageBoxDontAskAgainInterface_EnableMessage(@ptrCast(self), dontShowAgainName_str);
    }

    /// [Qt documentation](https://api.kde.org/kmessageboxdontaskagaininterface.html#enableMessage)
    ///
    /// Allows for overriding the related default method
    ///
    /// ``` self: QtC.KMessageBoxDontAskAgainInterface, slot: fn (self: QtC.KMessageBoxDontAskAgainInterface, dontShowAgainName: [*:0]const u8) callconv(.c) void ```
    pub fn OnEnableMessage(self: ?*anyopaque, slot: fn (?*anyopaque, [*:0]const u8) callconv(.c) void) void {
        qtc.KMessageBoxDontAskAgainInterface_OnEnableMessage(@ptrCast(self), @as(isize, @bitCast(@intFromPtr(&slot))));
    }

    /// [Qt documentation](https://api.kde.org/kmessageboxdontaskagaininterface.html#enableMessage)
    ///
    /// Base class method implementation
    ///
    /// ``` self: QtC.KMessageBoxDontAskAgainInterface, dontShowAgainName: []const u8 ```
    pub fn QBaseEnableMessage(self: ?*anyopaque, dontShowAgainName: []const u8) void {
        const dontShowAgainName_str = qtc.libqt_string{
            .len = dontShowAgainName.len,
            .data = dontShowAgainName.ptr,
        };
        qtc.KMessageBoxDontAskAgainInterface_QBaseEnableMessage(@ptrCast(self), dontShowAgainName_str);
    }

    /// [Qt documentation](https://api.kde.org/kmessageboxdontaskagaininterface.html#setConfig)
    ///
    /// ``` self: QtC.KMessageBoxDontAskAgainInterface, config: QtC.KConfig ```
    pub fn SetConfig(self: ?*anyopaque, config: ?*anyopaque) void {
        qtc.KMessageBoxDontAskAgainInterface_SetConfig(@ptrCast(self), @ptrCast(config));
    }

    /// [Qt documentation](https://api.kde.org/kmessageboxdontaskagaininterface.html#setConfig)
    ///
    /// Allows for overriding the related default method
    ///
    /// ``` self: QtC.KMessageBoxDontAskAgainInterface, slot: fn (self: QtC.KMessageBoxDontAskAgainInterface, config: QtC.KConfig) callconv(.c) void ```
    pub fn OnSetConfig(self: ?*anyopaque, slot: fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.KMessageBoxDontAskAgainInterface_OnSetConfig(@ptrCast(self), @as(isize, @bitCast(@intFromPtr(&slot))));
    }

    /// [Qt documentation](https://api.kde.org/kmessageboxdontaskagaininterface.html#setConfig)
    ///
    /// Base class method implementation
    ///
    /// ``` self: QtC.KMessageBoxDontAskAgainInterface, config: QtC.KConfig ```
    pub fn QBaseSetConfig(self: ?*anyopaque, config: ?*anyopaque) void {
        qtc.KMessageBoxDontAskAgainInterface_QBaseSetConfig(@ptrCast(self), @ptrCast(config));
    }

    /// [Qt documentation](https://api.kde.org/kmessageboxdontaskagaininterface.html#operator-eq)
    ///
    /// ``` self: QtC.KMessageBoxDontAskAgainInterface, param1: QtC.KMessageBoxDontAskAgainInterface ```
    pub fn OperatorAssign(self: ?*anyopaque, param1: ?*anyopaque) void {
        qtc.KMessageBoxDontAskAgainInterface_OperatorAssign(@ptrCast(self), @ptrCast(param1));
    }

    /// [Qt documentation](https://api.kde.org/kmessageboxdontaskagaininterface.html#dtor.KMessageBoxDontAskAgainInterface)
    ///
    /// Delete this object from C++ memory.
    ///
    /// ``` self: QtC.KMessageBoxDontAskAgainInterface ```
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.KMessageBoxDontAskAgainInterface_Delete(@ptrCast(self));
    }
};
